pulsar-metadata/src/test/java/org/apache/pulsar/metadata/ZKSessionTest.java

```diff
@@ -171,10 +168,12 @@ public void testReacquireLeadershipAfterSessionLost() throws Exception {
         e = sessionEvents.poll(10, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.SessionLost);
 
-        Awaitility.await().untilAsserted(() -> {
-            assertEquals(le1.getState(), LeaderElectionState.Leading);
-        });
-
+        // due to zookeeper async notice, we need to wait.
+        Awaitility.await()
+                .until(() -> le1.getState() == LeaderElectionState.Leading);
+        assertEquals(le1.getState(), LeaderElectionState.Leading);
+        Awaitility.await()
+                .until(()-> leaderElectionEvents.poll() == null);
```

Comment:
This seems to just make the Awaitility loop poll the elements in the queue, and it will definitely be null in the end. Did not really find the reason why the element is not empty.
Maybe should use peek instead of poll here

Reply:
Yes, I also found this problem just now, I will fix it. thanks to you~ :)

Reply:
I have one more question:
Why is leaderElectionEvents not null here? 
Even if it is changed to peek, it seems that it cannot be explained

Reply:
I don't think we need to test that LeaderElectionEvents is equal to null, because when the session is disconnected, the node will experience the noleader -> leader/Following state unless the state has not been updated when we get the leader state.

Reply:
What do you think about it?